edx/edx-enterprise
# [ENT-312] Require login for course-specific consent views

**Description:**

Right now, we raise a 404 when POSTing to a course-specific consent view, and, in fact, don't raise anything at all when GETing a course-specific consent view. In general, there aren't many valid flows that'll lead to this, but we still want to be robust about it. This pull request raises a 302 to the login screen when either GETing to the consent view or POSTing to the consent view.

**JIRA:** [ENT-312](https://openedx.atlassian.net/browse/ENT-312)

**Dependencies:** None

**Merge deadline:** 24 April 2017

**Installation instructions:** Basic enterprise installation

**Testing instructions:**

1. Set up an EnterpriseCustomer, enable consent, and set the consent requirement level to "at enrollment".
2. Add a user to the EnterpriseCustomer manually using the Manage Learners admin panel, while simultaneously enrolling the user in a course.
3. In the dashboard, click on the course and verify the consent interstitial appears. Make a note of the full URL of the interstitial.
4. Leaving that view alone, open a new tab and log out of your LMS session.
5. Back in the original tab, decline consent; note that you're redirected to the login page.
6. Upon logging in, verify that you're returned to the consent page. Verify that the URL is the same as the one you noted earlier.
7. Decline consent to return to the Dashboard.
8. Open a new tab and log out of your LMS session.
9. In the original tab, click on the course.
10. Verify that you're redirected to the login page.
11. Log in, and verify that you're redirected to the consent page. Again, verify that the URL matches the one you noted earlier.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**
